jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/WebSocketConnection.java

```diff
@@ -572,6 +599,7 @@ void enqueueFrame(Frame frame, Callback callback, boolean batch)
         private Flusher(Scheduler scheduler, int bufferSize, Generator generator, EndPoint endpoint)
         {
             super(bufferPool, scheduler, generator, endpoint, bufferSize, 8);
+            setUseDirectByteBuffers(isUseOutputDirectByteBuffers());
```

Comment:
Not sure I follow this one.
Seems redundant.

Reply:
The base class, `FrameFlusher`, has a `batchBuffer` that needs to be created heap or direct.
However, only here in the actual connection we can retrieve the heap/direct configuration, and fortunately the `FrameFlusher` was already subclassed here, so we set the directness on the flusher with the value from the connection.

Reply:
Ah, I understand now.